monkey/monkey_island/cc/services/initialize.py

```diff
@@ -100,6 +108,32 @@ def _register_conventions(container: DIContainer):
     )
 
 
+def _register_event_queues(container: DIContainer):
+    event_queue_lock = threading.Lock()
+
+    agent_event_queue = container.resolve(PyPubSubAgentEventQueue)
+    decorated_agent_event_queue = _decorate_agent_event_queue(agent_event_queue, event_queue_lock)
+    container.register_instance(IAgentEventQueue, decorated_agent_event_queue)
+
+    island_event_queue = container.resolve(PyPubSubIslandEventQueue)
+    decorated_island_event_queue = _decorate_island_event_queue(
+        island_event_queue, event_queue_lock
```

Comment:
Since these are two separate queues, we could use two separate locks

Reply:
If agent events occur simultaneously with certain island events (e.g. clear simulation data), it's important for them run sequentially to avoid race conditions.

Reply:
https://github.com/guardicore/monkey/pull/2374/commits/82e08ba157cc051e080a5bcb4d67ec410ac56257